### ISSUE TITLE
Collect and identify scenarios for E2E tests

### ISSUE THREAD

**Issue by aalexandrov** [MaterializeInc/materialize#17912]
Source a small representative set of E2E test scenarios to guide us through the WMR stabilization effort.

Possible sources are:
* ~LDBC queries~ (blocked by #17591)
* Workloads inspired from prospect beta testers.
* ~E2E tests from similar systems.~ (not now)

See the "Testing and observability" section from the design doc (#17820) for context.

### LDBC

* https://github.com/ldbc/ldbc_snb_bi/tree/main/umbra
* https://github.com/ldbc/ldbc_snb_datagen_spark#docker-images
* https://github.com/ldbc/ldbc_snb_bi/blob/main/snb-bi-pre-generated-data-sets.md
* https://ldbcouncil.org/ldbc_snb_datagen_spark/
* Queries
  * [ldbc/ldbc_snb_bi@main/umbra/queries/bi-15.sql](https://github.com/ldbc/ldbc_snb_bi/blob/main/umbra/queries/bi-15.sql?rgh-link-date=2023-03-08T11%3A49%3A13Z)
  * [ldbc/ldbc_snb_bi@main/umbra/queries/bi-19.sql](https://github.com/ldbc/ldbc_snb_bi/blob/main/umbra/queries/bi-19.sql?rgh-link-date=2023-03-08T11%3A49%3A13Z)
  * [ldbc/ldbc_snb_bi@main/umbra/queries/bi-20.sql](https://github.com/ldbc/ldbc_snb_bi/blob/main/umbra/queries/bi-20.sql?rgh-link-date=2023-03-08T11%3A49%3A13Z)

### Reachability for Introspection

* [See the design doc appendix](https://github.com/aalexandrov/materialize/blob/issue_17012/doc/developer/design/20230223_stabilize_with_mutually_recursive.md#appendix-internal-use-cases)

### Session windows

* [See this comment in #8698](https://github.com/MaterializeInc/materialize/issues/8698#issuecomment-1442162392)

### Tree aggregations

* Given a tree defined by an adjacency matrix, and some changing observations associated with the leafs, define a recursive query to aggregate these observations for every inner node of the tree. Define another query that operates on these aggregations (for example, to filter out values that are not in the expected range).

**Comment by vmarcos** [comment]
AFAICT, these are the queries from Umbra's SQL-based implementation of LDBC SNB BI that use `with recursive`:

- https://github.com/ldbc/ldbc_snb_bi/blob/main/umbra/queries/bi-15.sql
- https://github.com/ldbc/ldbc_snb_bi/blob/main/umbra/queries/bi-19.sql
- https://github.com/ldbc/ldbc_snb_bi/blob/main/umbra/queries/bi-20.sql

These could be interesting candidates to analyze and think how to express with `WMR` for scalability / incremental computation tests, given the benchmark's data generation tools.

**Comment by mgree** [comment]
For clarity/posterity: I'm working on [LDBC BI](https://github.com/ldbc/ldbc_snb_bi/tree/main), a workload with an initial snapshot and then several batches inserts/updates/deletes. Several queries (and even some fundamental data definitions) use WMR.
